ArduinoCode/Driver-IO/dashboard/dashboard.ino

```diff
@@ -0,0 +1,148 @@
+// Author: Joshua Cheng
+// Date: November 3, 2020
+
+#include <mcp_can.h> // uses seeed-studio's CAN_BUS_Shield library
+#include <mcp_can_dfs.h>
+#include <SPI.h>
+
+const int spiCSPin = 10; // Base CAN pin
+MCP_CAN CAN(spiCSPin); // Set CS pin
+
+const unsigned char ON[] = {1}; // Message for turning inverter on
+const unsigned char OFF[] = {0}; // Message for turning inverter off
+const unsigned char FORWARD[] = {1}; // Message for setting inverter to forward
+const unsigned char REVERSE[] = {0}; // Message for setting inverter to reverse
+
+const int SHUTDOWN_DETECTION_PIN = 2; // Pin for detecting shutdown system voltage.
+const int BUTTON_PIN = 3; // Pin for on/off button.
+const int GEAR_SELECTOR_PIN = 4; // Pin for forward/reverse selector
+const int IMD_INPUT_PIN = 5; // Pin for detecting IMD status
+const int BMS_INPUT_PIN = 6; // Pin for detecting BMS status
+
+const int IMD_INDICATOR_PIN = 8; // Pin for IMD indicator
+const int BMS_INDICATOR_PIN = 9; // Pin for BMS indicator
+const int BUTTON_LED_PIN = 11; // Pin for integrated LED on the button.
+const int SPEAKER_PIN = 10; // Pin for speaker or relay for speaker (must be pwm pin).
+
+bool shutdownVoltage = false;
+unsigned long speakerCooldown = 0; // Use unsigned long to match millis() function. Will overflow in around 50 days.
+unsigned long buttonCooldown = 0; // Use unsigned long to match millis() function. Will overflow in around 50 days.
+bool state = false;
+bool prevDirection = false;
+
+/**
+    Sets relavent pins to output or input, initializes serial for debugging, initializes CAN.
+*/
+void setup() {
+  Serial.begin(115200);
+
+  pinMode(BUTTON_PIN, INPUT_PULLUP); // Uses built-in pullup resistor.
+  pinMode(GEAR_SELECTOR_PIN, INPUT_PULLUP); // Uses built-in pullup resistor.
+
+  pinMode(IMD_INDICATOR_PIN, OUTPUT);
+  pinMode(BMS_INDICATOR_PIN, OUTPUT);
+  pinMode(SPEAKER_PIN, OUTPUT);
+  pinMode(BUTTON_LED_PIN, OUTPUT);
+
+  while (CAN_OK != CAN.begin(CAN_500KBPS, MCP_8MHz)) { //specify 8MHz crystal
+    Serial.println("CAN BUS init Failed");
+    delay(250);
+  }
+  Serial.println("CAN BUS Shield Init OK!");
+}
+
+void imdDetection();
+
+void bmsDetection();
+
+bool forwardReverse();
+
+void onOff();
+
+void changeDireciton();
+
+void loop() {
+  imdDetection();
+  bmsDetection();
+  onOff();
+  changeDirection();
+}
+
+/**
+   Turns IMD indicator on or off depending on if IMD_INPUT_PIN detects a voltage.
+*/
+void imdDetection() {
+  if (digitalRead(IMD_INPUT_PIN)) {
+    digitalWrite(IMD_INDICATOR_PIN, HIGH);
+  } else {
+    digitalWrite(IMD_INDICATOR_PIN, LOW);
+  }
+}
+
```

Comment:
These two methods could technically be refactored for an abstraction, whereby they each call a 3rd function and pass an input and an output, but this not required by any means and @mccauleyma will probably say it's very unnecessary. That said, reduction of duplicate code is a principle of good software development. I've provided the example below primarily for your learning.
```
void pinDetection(input, output) {
   if (digitalRead(input)) {
     digitalWrite(output, HIGH);
   } else {
     digitalWrite(output, LOW);
   }
}

void imdDetection() {
   pinDetection(IMD_INPUT_PIN, IMD_INDICATOR_PIN);
}

void bmsDetection() {
   pinDetection(BMS_INPUT_PIN, BMS_INDICATOR_PIN);
}
```

Reply:
Yep, unnecessary. More importantly though this code will be deleted anyways: https://github.com/Northeastern-Electric-Racing/NER/pull/71#pullrequestreview-527498466